CODING_STYLE.md

```diff
@@ -0,0 +1,335 @@
+#### General Notes
+
+- The coding style guide refers to new code. But legacy code can be cleaned up
+  and we are happy to take those patches.
+- Just because there is still code in LXC that doesn't adhere to the coding
+  standards outlined here does not license not adhering to the coding style. In
+  other words: please stick to the coding style.
+
+#### Only Use Tabs
+
+- LXC uses tabs.
+
+#### Only use /* Style Comments */
+
+- Any comments that are added must use /* */.
+
+#### Try To Stick To Wrap At 80chars
+
+- This is not strictly enforced. It is perfectly legal to sometimes
+  overflow this limit if it helps clarity. Nonetheless, try to stick to it
+  and use common sense to decide when not to.
+
+#### Error Messages
+
+- Error messages must start with a capital letter and must **not** end with a
+  punctuation sign.
+- They should be descriptive, without being needlessly long. It is best to just
+  use already existing error messages as examples.
+- Examples of acceptable error messages are:
+  ```
+  SYSERROR("Failed to create directory \"%s\"", path);
+  WARN("\"/dev\" directory does not exist. Proceeding without autodev being set up");
+  ```
+
+#### Return Error Codes
+
+- When writing a function that can fail in a non-binary way try to return
+  meaningful negative error codes (e.g. `return -EINVAL;`).
+
+#### All Unexported Functions Must Be Declared `static`
+
+- Functions which are only used in the current file and are not exported
+  within the codebase need to be declared with the `static` attribute.
+
+#### All Names Must Be In lower_case
+
+- All functions and variable names must use lower case.
+
+#### Declaring Variables
+
+- variables should be declared at the top of the function or at the beginning
+  of a new scope but **never** in the middle of a scope
+- separate initialized from uninitialized variable declarations
+- put multiple declarations of the same type on the same line
+- standard types defined by libc should preceed types defined by LXC
```

Comment:
preceed->precede
but isn't the example below backwards then?

Reply:
I'm clarifying this.